python/pyspark/worker.py

```diff
@@ -500,7 +501,12 @@ def main(infile, outfile):
 
             except (resource.error, OSError, ValueError) as e:
                 # not all systems support resource limits, so warn instead of failing
-                print("WARN: Failed to set memory limit: {0}\n".format(e), file=sys.stderr)
+                print(warnings.formatwarning(
+                    "Failed to set memory limit: {0}\n".format(e),
+                    ResourceWarning,
+                    __file__,
+                    getframeinfo(currentframe()).lineno
```

Comment:
Ah, gotya. To be extremely conservative for the Python implementation concern, we could do something like:

```python
cur = currentframe()
getframeinfo(cur).lineno if cut is not None else -1  # or 0 ..?
```